### PR TITLE
Managed Identity changes

### PR DESCRIPTION
**Pull request by nirarg**
##### SUMMARY
This PR is finalize and cleanup work for Managed Identity feature.
This include the following changes:

1. Remove "update_identities" definition and make sure all modules use "update_managed_identity" instead
2. Remove "get new_identity" back off logic and make sure "new_identity" param is always provided ("new_identity" shouldn't be optional)
3. Move all managed identity specs to "azure_rm_common_ext.py" to avoid duplicated specs , also move the relevant identity options description in doc_fragments
4. For integration tests - consider using common tasks files for all managed identities preparation
6.  In integration tests - change all places using hard coded region to use '{{ location }}'

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
This Change is relevant for all modules that use the ManagedIdentity:

- plugins/modules/azure_rm_batchaccount.py
- plugins/modules/azure_rm_storageaccount.py
- plugins/modules/azure_rm_diskencryptionset.py
- plugins/modules/azure_rm_servicebus.py
- plugins/modules/azure_rm_recoveryservicesvault.py
- plugins/modules/azure_rm_cdnprofile.py
- plugins/modules/azure_rm_cosmosdbaccount.py
- plugins/modules/azure_rm_functionapp.py
- plugins/modules/azure_rm_eventhub.py
- plugins/modules/azure_rm_rediscache.py
- plugins/modules/azure_rm_appgateway.py
- plugins/modules/azure_rm_containerregistry.py
- plugins/modules/azure_rm_containerinstance.py
- plugins/modules/azure_rm_virtualmachinescaleset.py
- plugins/modules/azure_rm_sqlserver.py
- plugins/modules/azure_rm_postgresqlflexibleserver.py
- plugins/modules/azure_rm_webapp.py
- plugins/modules/azure_rm_iothub.py

### CODE REVIEW

**Comment by nirarg** [comment]
@Fred-sun @xuzhang3 @p3ck 
The PR is ready for review
All the modules mentioned in the description have been tested locally on my computer
What is the recommended way to run CI here? should it be executed separately for each test?

**Comment by Fred-sun** [comment]
@nirarg You can choose to execute all the test cases like the following, thank you!
image

**Comment by nirarg** [comment]
Thanks @Fred-sun 
I started the pipeline with "all" tests
The progress details and results can be found here:
https://dev.azure.com/azclitools/public/_build/results?buildId=188142&view=results

**Comment by Fred-sun** [comment]
@nirarg Some mistakes may occur. Recently, there was a policy to prohibit the use of port 22, and some of our use cases reported errors for this reason. I have fixed in another PR #1704. When they merge tomorrow(I will notice to you the result tomorrow.), they'll be ready for normal testing. Thank you!

**Review by p3ck — DISMISSED** [review]
LGTM!  Thanks

**Comment by Fred-sun** [comment]
@nirarg #1704  already merged! Thanks!

**Comment by nirarg** [comment]
> @nirarg #1704 already merged! Thanks!

Thank you @Fred-sun 
I rebased this PR on `dev` branch and re-executed the pipeline
Progress and results can be found here:
https://dev.azure.com/azclitools/public/_build/results?buildId=188891&view=results

**Comment by nirarg** [comment]
Hi @Fred-sun ,
The pipeline have finished with some errors, which I think that are not related to the changes in this PR
Can you please check this?

File: tests/integration/targets/azure_rm_appgateway/tasks/main.yml [hunk 1]
```diff
@@ -16,25 +16,18 @@
...
-    api_version: "2023-01-31"
-    body:
-      location: "{{ location }}"
-    state: present
-  loop:
-    - "ansible-test-appgw-identity"
-    - "ansible-test-appgw-identity-2"
-
-- name: Set identities IDs to test. Identities ansible-test-appgw-identity and ansible-test-appgw-identity-2 have to be created previously
+
```
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_appgateway/tasks/main.yml]: Delete trailing spaces!
```suggestion

```

Comment (nirarg) [reply tests/integration/targets/azure_rm_appgateway/tasks/main.yml]: Thank you, fixed

**Comment by nirarg** [comment]
Hi @Fred-sun ,
Is there any other action needed here before merge this PR?

**Comment by xuzhang3** [comment]
@nirarg Failed tests are the module changed by this PR: (https://dev.azure.com/azclitools/public/_build/results?buildId=190863&view=logs&j=74643a41-85da-5afc-0d1e-5e3c1770b297&t=469d1757-8389-5c48-bd4e-7213ae177b56&l=2987)

**Comment by Fred-sun** [comment]
Repeate with #1724(merged), Closed it!

**Pull request closed without merging**